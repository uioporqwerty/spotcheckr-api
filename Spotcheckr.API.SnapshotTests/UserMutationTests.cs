using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Spotcheckr.API.Tests.Common.Fixtures;
using Xunit;

namespace Spotcheckr.API.SnapshotTests
{
	public class UserMutationTests : BaseSnapshotTest
	{
		public UserMutationTests(ServiceFixture serviceFixture) : base(serviceFixture)
		{
		}

		[Fact]
		public async void CreateUser_WithPost_ReturnsNewUser()
		{
			var executor = await GetRequestExecutorAsync();
			var newUser = await executor.ExecuteAsync(@"
				mutation {
					createUser(input: { userType: ATHLETE }) {
						user {
							id
						}
					}
				}
			");
			newUser.ToJson().MatchSnapshot();
		}
	}
}
