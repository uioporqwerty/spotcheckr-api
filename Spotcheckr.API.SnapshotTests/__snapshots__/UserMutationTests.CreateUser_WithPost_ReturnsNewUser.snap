{
  "data": {
    "createUser": {
      "user": {
        "id": "QXRobGV0ZQppMQ=="
      }
    }
  }
}
